research/conditional/utils/conditional_trainer.py

```diff
@@ -134,7 +134,11 @@ def train(self, n_steps: int):
             if step % 1000 == 0:
                 print(f"Step {step}")
 
-            if self.model_type == "gpt" and step % self.decoding_logging_steps == 0:
+            if (
+                self.model_type == "gpt"
+                and step % self.decoding_logging_steps == 0
+                and self.logger is not None
```

Comment:
Może dodajmy komentarz, że tylko cuda:0 ma loggera, bo to nieoczywiste

Reply:
@crewtool  @kuba-krj a może dałoby się dodać parametr w rodzaju `self.is_logging_process`?

Reply:
Dodałem parametr